diff --git a/tools/server/server-context.cpp b/tools/server/server-context.cpp
index ad4eb3578..163571047 100644
--- a/tools/server/server-context.cpp
+++ b/tools/server/server-context.cpp
@@ -3722,7 +3722,7 @@ private:
                         //  - 4
                         // ref: https://github.com/ggml-org/llama.cpp/pull/20288
                         if (do_checkpoint) {
-                            static const int checkpoint_offsets[] = {4 + n_ubatch, 4};
+                            static const int checkpoint_offsets[] = {4 + n_ubatch, 64, 4};
 
                             bool should_break = false;
                             for (int offset : checkpoint_offsets) {
